src/lib/Zikula/Core/Helper/ExtensionVar.php

```diff
@@ -0,0 +1,283 @@
+     *
+     * @param string $extensionName The name of the extension.
+     * @param string $variableName The name of the variable.
+     *
+     * @return boolean True if the variable exists in the database, false if not.
+     */
+    public function has($extensionName, $variableName)
+    {
+        // define input, all numbers and booleans to strings
+        $extensionName = isset($extensionName) ? ((string) $extensionName) : '';
+        $variableName = isset($variableName) ? ((string) $variableName) : '';
+
+        // The cast to (array) is for the odd instance where $this->variables[$extensionName] is set to null--not sure if this is really needed.
+        $varExists = isset($this->variables[$extensionName]) && array_key_exists($variableName, (array)$this->variables[$extensionName]);
+
+        if (!$varExists && \System::isUpgrading()) {
+            // Handle upgrade edge case: a call to get() ensures vars for the extension are loaded if newly available.
+            $vars = $this->get($extensionName);
+            $varExists = array_key_exists($variableName, (array)$vars);
+        }
+
+        return $varExists;
+    }
+
+    /**
+     * Get an extension variable.
+     *
+     * If the variableName parameter is omitted the method returns a multi-dimensional array of the keys and values
+     * for all the extension vars.
+     *
+     * @param string $extensionName The name of the extension or pseudo-extension (e.g., 'ZikulaUsersModule', 'ZConfig', '/EventHandlers').
+     * @param string $variableName The name of the variable.
+     * @param mixed $default The value to return if the requested var is not set.
+     *
+     * @return string|array If the name parameter is included then method returns
+     *          string - extension variable value
+     *          if the name parameter is ommitted then method returns
+     *          array - multi-dimensional array of the keys and values for the extension vars.
+     */
+    public function get($extensionName, $variableName = '', $default = false)
+    {
+        if (empty($extensionName)) {
+            throw new \InvalidArgumentException();
+        }
+
+        // if vars for this extension not available then get them
+        if (!array_key_exists($extensionName, $this->variables)) {
+            // A query out to the database should only be needed if the system is upgrading. Use the installing flag to determine this.
+            if (\System::isUpgrading()) {
+                $this->initExtensionVars(true);
+            } else {
+                // Prevent a re-query for the same extension in the future, where the extension does not define any extension variables.
+                $this->variables[$extensionName] = array();
+            }
+        }
+
+        // if variable name not passed, then return every variable for the specified extension as an associative array.
+        if (empty($variableName) && array_key_exists($extensionName, $this->variables)) {
```

Comment:
Maybe extract this to a *getAll* method and disallow empty variable names for the get method.